### PR TITLE
Add build configuration to generate runtime packs for iOS

### PR DESCRIPTION
**Pull request by steveisok**
This adds support for generating iOS runtime packs, which is intended to be consumed by the Xamarin iOS SDK.

### CODE REVIEW

File: eng/Subsets.props [hunk 1]
```diff
@@ -218,15 +220,30 @@
...
     <PkgprojProjectToBuild Include="$(InstallerProjectRoot)pkg\projects\**\*.pkgproj" SignPhase="MsiFiles" BuildInParallel="false" />
     <ProjectToBuild Include="@(PkgprojProjectToBuild)" />
   </ItemGroup>
 
+  <ItemGroup Condition="$(_subsetCategory.Contains('installer')) and $(_subset.Contains('-pkgproj-')) and '$(TargetOS)' == 'iOS'">    
+    <!-- For iOS we only need to construct runtime packs-->
+    <PkgprojProjectToBuild  
+        Include="$(InstallerProjectRoot)pkg\projects\netcoreapp\pkg\Microsoft.NETCore.App.Runtime.pkgproj" 
+        SignPhase="MsiFiles" BuildInParallel="false" />
+    <PkgprojProjectToBuild
```
Comment (marek-safar) [inline eng/Subsets.props]: Could you add comment what legacy and workaround is about?

File: eng/Subsets.props [hunk 2]
```diff
@@ -58,8 +58,10 @@
 
   <PropertyGroup>
     <DefaultSubsetCategories>libraries-installer-coreclr-mono</DefaultSubsetCategories>
-    <DefaultSubsetCategories Condition="'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'Android'">libraries-mono</DefaultSubsetCategories>
+    <DefaultSubsetCategories Condition="'$(TargetOS)' == 'iOS'">libraries-installer-mono</DefaultSubsetCategories>
+    <DefaultSubsetCategories Condition="'$(TargetOS)' == 'Android'">libraries-mono</DefaultSubsetCategories>
     <DefaultInstallerSubsets>corehost-managed-depproj-pkgproj-bundle-installers-test</DefaultInstallerSubsets>
+    <DefaultInstallerSubsets Condition="'$(TargetOS)' == 'iOS'">managed-depproj-pkgproj-bundle-test</DefaultInstallerSubsets>
```
Comment (akoeplinger) [inline eng/Subsets.props]: why do we need anything other than the `depproj` and `pkgproj ` subsets?

File: eng/Subsets.props (hunk 1 above)
Comment (akoeplinger) [reply eng/Subsets.props]: Those are pre-existing projects/directories but yeah it'd be good to find out what they're used for and whether we can simplify.

File: eng/liveBuilds.targets [hunk 3]
```diff
@@ -122,8 +127,14 @@
...
+      <CoreCLRFiles>
+        <IsNative>true</IsNative>
+      </CoreCLRFiles>
+    </ItemGroup>
 
-    <Error Condition="'@(CoreCLRFiles)' == ''" Text="The CoreCLR subset category must be built before building this project." />
+    <Error Condition="'@(CoreCLRFiles)' == ''" Text="The $(RuntimeFlavor) subset category must be built before building this project." />
   </Target>
 
   <Target Name="ResolveMonoFilesFromLocalBuild">
```
Comment (mdh1418) [inline eng/liveBuilds.targets]: Do we still need this target?

Comment (mdh1418) [reply eng/liveBuilds.targets]: If we also change `ResolveMonoFilesFromLocalBuild` to `ResolveRuntimeFilesFromLocalBuild` or even a more concise way to combine the two in `eng/referenceFromRuntime/targets`?

File: eng/Subsets.props [hunk 4]
```diff
@@ -218,15 +220,30 @@
...
     <ProjectToBuild Include="@(DepprojProjectToBuild)" />
   </ItemGroup>
 
-  <ItemGroup Condition="$(_subsetCategory.Contains('installer')) and $(_subset.Contains('-pkgproj-'))">
+  <ItemGroup Condition="$(_subsetCategory.Contains('installer')) and $(_subset.Contains('-pkgproj-')) and '$(TargetOS)' != 'iOS'">
     <PkgprojProjectToBuild Include="$(InstallerProjectRoot)pkg\projects\**\*.pkgproj" SignPhase="MsiFiles" BuildInParallel="false" />
     <ProjectToBuild Include="@(PkgprojProjectToBuild)" />
   </ItemGroup>
 
+  <ItemGroup Condition="$(_subsetCategory.Contains('installer')) and $(_subset.Contains('-pkgproj-')) and '$(TargetOS)' == 'iOS'">
```
Comment (dagood) [inline eng/Subsets.props]: My initial impression is that I don't like seeing this complexity in the subsets system. It seems more maintainable and readable (discoverable) to me for the pkgproj/depproj/etc. to be able to do the right thing when encountering this new TargetOS.

File: eng/liveBuilds.targets [hunk 5]
```diff
@@ -78,10 +78,11 @@
     <CoreCLRCrossTargetComponentDirName Condition="'$(TargetArchitecture)' == 'arm' and '$(BuildArchitecture)' != 'arm' and '$(TargetsLinux)' == 'true'">x64</CoreCLRCrossTargetComponentDirName>
   </PropertyGroup>
 
-  <Target Name="ResolveCoreCLRFilesFromLocalBuild">
-    <Error Condition="!Exists('$(CoreCLRArtifactsPath)')" Text="The CoreCLR artifacts path does not exist '$(CoreCLRArtifactsPath)'. The CoreCLR subset category must be built before building this project." />
+  <Target Name="ResolveRuntimeFilesFromLocalBuild">
```
Comment (dagood) [inline eng/liveBuilds.targets]: Should we say `VM` or something better, to avoid being ambiguous between "CoreCLR vs. Mono" and "the Runtime repository"?

File: eng/pipelines/installer/jobs/base-job.yml [hunk 6]
```diff
@@ -186,7 +186,6 @@ jobs:
       value: >-
         --restore --build --ci --test
         /p:CrossBuild=${{ ne(parameters.crossrootfsDir, '') }}
-        /p:DisableCrossgen=$(_DisableCrossgen)
```
Comment (dagood) [inline eng/pipelines/installer/jobs/base-job.yml]: Clean up unused variable earlier in file?

File: src/installer/Directory.Build.props [hunk 7]
```diff
@@ -129,7 +130,7 @@
   <PropertyGroup>
     <DisableCrossgen>false</DisableCrossgen>
     <!-- Disable cross-gen on FreeBSD for now. This can be revisited when we have full support. -->
-    <DisableCrossgen Condition="'$(TargetOS)'=='FreeBSD'">true</DisableCrossgen>
+    <DisableCrossgen Condition="'$(TargetOS)'=='FreeBSD' or '$(RuntimeFlavor)' == 'Mono'">true</DisableCrossgen>
```
Comment (dagood) [inline src/installer/Directory.Build.props]: Put on separate line to simplify FreeBSD fix later, and clarify the target of the comment?

File: src/installer/Directory.Build.props [hunk 8]
```diff
@@ -147,10 +148,10 @@
...
-    <InstallerStartSuffix Condition="'$(TargetOS)' == 'OSX'">internal</InstallerStartSuffix>
-    <SharedHostInstallerStart Condition="'$(TargetOS)' == 'OSX'">$(SharedHostInstallerStart)$(InstallerStartSuffix)-</SharedHostInstallerStart>
-    <HostFxrInstallerStart Condition="'$(TargetOS)' == 'OSX'">$(HostFxrInstallerStart)$(InstallerStartSuffix)-</HostFxrInstallerStart>
-    <SharedFrameworkInstallerStart Condition="'$(TargetOS)' == 'OSX'">$(SharedFrameworkInstallerStart)$(InstallerStartSuffix)-</SharedFrameworkInstallerStart>
+    <InstallerStartSuffix Condition="'$(TargetOS)' == 'OSX' or '$(TargetOS)' == 'iOS'">internal</InstallerStartSuffix>
+    <SharedHostInstallerStart Condition="'$(TargetOS)' == 'OSX' or '$(TargetOS)' == 'iOS'">$(SharedHostInstallerStart)$(InstallerStartSuffix)-</SharedHostInstallerStart>
+    <HostFxrInstallerStart Condition="'$(TargetOS)' == 'OSX' or '$(TargetOS)' == 'iOS'">$(HostFxrInstallerStart)$(InstallerStartSuffix)-</HostFxrInstallerStart>
+    <SharedFrameworkInstallerStart Condition="'$(TargetOS)' == 'OSX' or '$(TargetOS)' == 'iOS'">$(SharedFrameworkInstallerStart)$(InstallerStartSuffix)-</SharedFrameworkInstallerStart>
```
Comment (dagood) [inline src/installer/Directory.Build.props]: We don't plan to produce shared framework installers for iOS, do we? (Add to comment about this? The change itself seems harmless either way but it stands out and could be confusing.)

File: src/installer/Directory.Build.props [hunk 9]
```diff
@@ -276,7 +284,7 @@
     <CompressedFileExtension Condition="'$(TargetOS)' == 'Windows_NT'">.zip</CompressedFileExtension>
     <CompressedFileExtension Condition="'$(TargetOS)' != 'Windows_NT'">.tar.gz</CompressedFileExtension>
     <InstallerExtension Condition="'$(TargetOS)' == 'Windows_NT'">.msi</InstallerExtension>
-    <InstallerExtension Condition="'$(TargetOS)' == 'OSX'">.pkg</InstallerExtension>
+    <InstallerExtension Condition="'$(TargetOS)' == 'OSX' or '$(TargetOS)' == 'iOS'">.pkg</InstallerExtension>
```
Comment (dagood) [inline src/installer/Directory.Build.props]: (Similar as above) Are we really producing a sharedfx pkg for iOS?

File: src/installer/pkg/packaging/osx/package.targets [hunk 10]
```diff
@@ -5,13 +5,13 @@
   <UsingTask TaskName="ReplaceFileContents" AssemblyFile="$(InstallerTasksAssemblyPath)"/>
 
   <Target Name="InitPkg"
-          Condition="'$(TargetOS)' == 'OSX'">
+          Condition="'$(TargetOS)' == 'OSX' or '$(TargetOS') == 'iOS'">
```
Comment (dagood) [inline src/installer/pkg/packaging/osx/package.targets]: Typo, quote misplaced.

File: src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj [hunk 11]
```diff
@@ -4,7 +4,7 @@
     Include HostPolicy and HostResolver based on legacy packages. This can be collapsed once the
     legacy packages are no longer used.
   -->
-  <Import Project="..\..\Microsoft.NETCore.DotNetHostPolicy\Microsoft.NETCore.DotNetHostPolicy.props" />
-  <Import Project="..\..\Microsoft.NETCore.DotNetHostResolver\Microsoft.NETCore.DotNetHostResolver.props" />
+  <Import Condition="'$(TargetOS)' != 'iOS'" Project="..\..\Microsoft.NETCore.DotNetHostPolicy\Microsoft.NETCore.DotNetHostPolicy.props" />
+  <Import Condition="'$(TargetOS)' != 'iOS'" Project="..\..\Microsoft.NETCore.DotNetHostResolver\Microsoft.NETCore.DotNetHostResolver.props" />
```
Comment (dagood) [inline src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj]: Is iOS really what matters here? I would expect Mono (*if* we don't want to include the host resolver or host policy libraries with it).

File: eng/Subsets.props (hunk 4 above)
Comment (dagood) [reply eng/Subsets.props]: For example, pkgproj Build has functionality to allow a complete skip:

https://github.com/dotnet/arcade/blob/7c5a68c1aeafec3a9a1b63d1d2c3b197964cc1b7/src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/framework.packaging.targets#L237-L239

A potential way is to manipulate `BuildOnUnknownPlatforms` to trigger a skip based on RID not being present:

https://github.com/dotnet/runtime/blob/1d018ba7e9bfcbc8eb1f56fd14532e7fcfc480c1/src/installer/pkg/projects/Directory.Build.props#L117-L119

https://github.com/dotnet/arcade/blob/866b2acd5ffc3c2031c102f6415fd7c6a1a370d5/src/Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk/targets/packaging-tools.targets#L167-L168

File: eng/liveBuilds.targets (hunk 3 above)
Comment (steveisok) [reply eng/liveBuilds.targets]: I wasn't sure how big of a bite to take.  We would take away `ResolveMonoFilesFromLocalBuild` and probably want to change `<CoreCLRFiles>` too

File: src/installer/Directory.Build.props (hunk 9 above)
Comment (steveisok) [reply src/installer/Directory.Build.props]: No, I was taking a scorched earth approach and can take it away.

File: src/installer/Directory.Build.props (hunk 8 above)
Comment (steveisok) [reply src/installer/Directory.Build.props]: No, I'll take it away.

File: src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj (hunk 11 above)
Comment (steveisok) [reply src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj]: I made it iOS here because mono should also be able to generate the same things as coreclr.  Generating just runtime packs are only going to apply to iOS and Android.

File: eng/Subsets.props (hunk 1 above)
Comment (steveisok) [reply eng/Subsets.props]: We may not need it if I can handle it at the pkgproj level.

@dagood Do you know the history?

Comment (dagood) [reply eng/Subsets.props]: Legacy produces the old nupkgs that formed NETCoreApp in 2.1. Workaround produces a package to work around some issues that were found in 3.0 SDK development. https://github.com/dotnet/core-setup/pull/6096, https://github.com/dotnet/runtime/issues/3528.

But yes, I do not think they should be called out explicitly in this file. Adding a readme to the project dir or inside the project files would be the way to go IMO.

Comment (dagood) [reply eng/Subsets.props]: > whether we can simplify.

Probably. However, the Host tests (`src/installer/test/Assets`) currently still depend on the legacy packages. The legacy packages were also used for testing by downstream repos even after 3.0, and may still be.

File: src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj (hunk 11 above)
Comment (dagood) [reply src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Runtime.pkgproj]: I think it'd be nice to make this clear by adding a property and clarifying comment (that doesn't interfere with existing comment):

```xml
<!--
  Don't include host resolver or host policy in iOS or Android runtime packs, because there is no
  shared framework on those platforms.
-->
<PropertyGroup>
  <ExcludeHostAssets Condition="'$(TargetOS)' == 'iOS'">true</ExcludeHostAssets>
  <ExcludeHostAssets Condition="'$(TargetOS)' == 'Android'">true</ExcludeHostAssets>
</PropertyGroup>

<!--
  Include HostPolicy and HostResolver based on legacy packages. This can be collapsed once the
  legacy packages are no longer used.
-->
<Import Condition="'$(ExcludeHostAssets)' != 'true'" Project="..\..\Microsoft.NETCore.DotNetHostPolicy\Microsoft.NETCore.DotNetHostPolicy.props" />
<Import Condition="'$(ExcludeHostAssets)' != 'true'" Project="..\..\Microsoft.NETCore.DotNetHostResolver\Microsoft.NETCore.DotNetHostResolver.props" />
```

File: eng/referenceFromRuntime.targets [hunk 12]
```diff
@@ -45,7 +45,7 @@
   <Target Name="GetFilesFromCoreCLR" 
           Condition="'$(RuntimeFlavor)' != 'Mono'"
           Returns="@(CoreCLRFiles)"
-          DependsOnTargets="ResolveCoreCLRFilesFromLocalBuild" />
+          DependsOnTargets="ResolveRuntimeFilesFromLocalBuild" />
```
Comment (jaredpar) [inline eng/referenceFromRuntime.targets]: Why did you change this target but not the `GetFilesFromMono` below? Those targets should mirror each other in this respect.

Comment (steveisok) [reply eng/referenceFromRuntime.targets]: I didn't want to change too much until after I got something working.  Now that I look again, I don't think it's a big deal to alter.

File: eng/liveBuilds.targets [hunk 13]
```diff
@@ -78,10 +78,11 @@
     <CoreCLRCrossTargetComponentDirName Condition="'$(TargetArchitecture)' == 'arm' and '$(BuildArchitecture)' != 'arm' and '$(TargetsLinux)' == 'true'">x64</CoreCLRCrossTargetComponentDirName>
   </PropertyGroup>
 
-  <Target Name="ResolveCoreCLRFilesFromLocalBuild">
-    <Error Condition="!Exists('$(CoreCLRArtifactsPath)')" Text="The CoreCLR artifacts path does not exist '$(CoreCLRArtifactsPath)'. The CoreCLR subset category must be built before building this project." />
+  <Target Name="ResolveVMFilesFromLocalBuild">
```
Comment (dagood) [inline eng/liveBuilds.targets]: I threw this suggestion out there without much context, so I want to highlight it for other reviewers... "runtime" as CoreCLR/Mono vs. "runtime" as the marketing name of the shared framework has been ambiguous for a while, and at some point early in Mono discussions "VM" came up as a possible less ambiguous name for the CoreCLR/Mono grouping. (The implications of "VM" might actually make it a straight-up incorrect suggestion here, I seem to remember some issue like that with the name, e.g. that aot tooling is included so it doesn't make sense.)

The code has already built up concepts like `RuntimeFlavor` which use the first definition, so maybe it's too late and doesn't matter enough to really change--context clues may be enough.

Comment (steveisok) [reply eng/liveBuilds.targets]: For me, either are fine.  I honestly thought runtime fit better, but I'll go with whatever the group decides.

File: eng/liveBuilds.targets [hunk 14]
```diff
@@ -78,10 +78,11 @@
...
+    <Error Condition="!Exists('$(CoreCLRArtifactsPath)') and '$(RuntimeFlavor)' == 'CoreCLR'" Text="The CoreCLR artifacts path does not exist '$(CoreCLRArtifactsPath)'. The CoreCLR subset category must be built before building this project." />
+    <Error Condition="!Exists('$(MonoArtifactsPath)') and '$(RuntimeFlavor)' == 'Mono'" Text="The Mono artifacts path does not exist '$(MonoArtifactsPath)'. The Mono subset category must be built before building this project." />
 
-    <PropertyGroup>
+    <PropertyGroup Condition="'$(RuntimeFlavor)' == 'CoreCLR'">
```
Comment (mdh1418) [inline eng/liveBuilds.targets]: Would it be clearer if we group the Property/Item groups by their conditions?
```
<Choose>
   <When Condition="'$(RuntimeFlavor)' == 'CoreCLR'">
      <PropertyGroup></PropertyGroup>
      <ItemGroup></ItemGroup>
   </When>
   <When Condition="'$(RuntimeFlavor)' == 'Mono'">
      <PropertyGroup></PropertyGroup>
      <ItemGroup></ItemGroup>
   </When>
</Choose>
```

File: src/installer/Directory.Build.props [hunk 15]
```diff
@@ -128,8 +129,9 @@
 
   <PropertyGroup>
     <DisableCrossgen>false</DisableCrossgen>
+    <DisableCrossgen Condition="'$(RuntimeFlavor)' == 'Mono'">true</DisableCrossgen>
     <!-- Disable cross-gen on FreeBSD for now. This can be revisited when we have full support. -->
-    <DisableCrossgen Condition="'$(TargetOS)'=='FreeBSD'">true</DisableCrossgen>
+    <DisableCrossgen Condition="'$(TargetOS)'=='FreeBSD'">true</DisableCrossgen>
```
Comment (mdh1418) [inline src/installer/Directory.Build.props]: remove extra space

File: src/installer/Directory.Build.props [hunk 16]
```diff
@@ -93,6 +93,7 @@
   <PropertyGroup Condition="'$(PortableBuild)' == 'true'">
     <OutputRid Condition="'$(TargetOS)' == 'Windows_NT'">win-$(TargetArchitecture)</OutputRid>
     <OutputRid Condition="'$(TargetOS)' == 'OSX'">osx-$(TargetArchitecture)</OutputRid>
+    <OutputRid Condition="'$(TargetOS)' == 'iOS'">iOS-$(TargetArchitecture)</OutputRid>
```
Comment (mdh1418) [inline src/installer/Directory.Build.props]: lowercase for consistency? Or is the iOS Rid prefix already used elsewhere?

File: eng/pipelines/runtime.yml [hunk 17]
```diff
@@ -481,6 +481,18 @@ jobs:
       liveRuntimeBuildConfig: release
       liveLibrariesBuildConfig: ${{ variables.debugOnPrReleaseOnRolling }}
 
+- template: /eng/pipelines/installer/installer-matrix.yml
```
Comment (safern) [inline eng/pipelines/runtime.yml]: This is only adding the legs to CI, if we want the runtime packages to be published, this needs to be added to runtime-official.yml as well.

File: eng/pipelines/installer/jobs/base-job.yml [hunk 18]
```diff
@@ -112,7 +112,7 @@ jobs:
         $(CommonMSBuildArgs)
         $(MsbuildSigningArguments)
 
-  - ${{ if eq(parameters.osGroup, 'OSX') }}:
+  - ${{ if or(eq(parameters.osGroup, 'OSX'), eq(parameters.osGroup, 'iOS')) }}:
```
Comment (safern) [inline eng/pipelines/installer/jobs/base-job.yml]: You can simplify this condition by using `in` conditional expression.

```yml
${{ if in(parameters.osGroup, 'OSX', 'iOS') }}:
```

That way you don't need to do an or of eq statements.

File: eng/Subsets.props (hunk 4 above)
Comment (steveisok) [reply eng/Subsets.props]: So, I skipped at the pkgproj level.  It worked, but it still generated an empty nupkg for each project.  It's unclear if I can actually skip that step.  Do you know?

File: src/libraries/restore/runtime/runtime.depproj [hunk 19]
```diff
@@ -74,27 +74,27 @@
   </Target>
 
   <Target Name="OverrideRuntimeCoreCLR"
```
Comment (jaredpar) [inline src/libraries/restore/runtime/runtime.depproj]: If we're unifying the `Resolve*FilesFromLocalBuild` should we unify the `OverrideRuntime*` target as well?

File: eng/liveBuilds.targets (hunk 13 above)
Comment (jaredpar) [reply eng/liveBuilds.targets]: Small preference for `ResolveRuntimeFilesFromLocalBuild` and `RuntimeFiles` because I feel like it has consistency with other property names / item groups we use today. Don't feel particularly strongly about it though.

File: eng/Subsets.props (hunk 4 above)
Comment (dagood) [reply eng/Subsets.props]: Yes, disabling `BuildOnUnknownPlatforms` prevents the WindowsDesktop packages from building where they shouldn't in Core-Setup release/3.1. Wouldn't expect to see empty nupkgs in any situation I know about.

File: eng/liveBuilds.targets [hunk 20]
```diff
@@ -91,26 +92,30 @@
...
         Condition="'$(CoreCLRCrossTargetComponentDirName)' != ''">$([MSBuild]::NormalizeDirectory('$(CoreCLRArtifactsPath)','$(CoreCLRCrossTargetComponentDirName)','sharedFramework'))</CoreCLRCrossTargetComponentDir>
     </PropertyGroup>
-    <ItemGroup>
-      <CoreCLRFiles Include="$(CoreCLRSharedFrameworkDir)*.*" />
+    <PropertyGroup Condition="'$(RuntimeFlavor)' == 'Mono'">
+      <MonoArtifactsPath>$([MSBuild]::NormalizeDirectory('$(MonoArtifactsPath)'))</MonoArtifactsPath>
+    </PropertyGroup>
+
+    <ItemGroup Condition="'$(RuntimeFlavor)' == 'CoreCLR'">
+      <VMFiles Include="$(CoreCLRSharedFrameworkDir)*.*" />
```
Comment (safern) [inline eng/liveBuilds.targets]: `RuntimeFiles` to be consistent with properties like `runtimeConfiguration` and `RuntimeFlavor`?

File: eng/liveBuilds.targets (hunk 13 above)
Comment (steveisok) [reply eng/liveBuilds.targets]: I prefer RuntimeFiles.  Who gets to decide?

Comment (dagood) [reply eng/liveBuilds.targets]: I put the dotnet/runtime-infrastructure team as a reviewer to discuss/reject the name. I believe the general model is that it's up to your judgment unless repo or area owners care otherwise. So far it sounds like consistency in property naming + not overhauling properties during this PR outweighs the ambiguity, and I don't disagree with that.

File: src/libraries/restore/runtime/runtime.depproj (hunk 19 above)
Comment (steveisok) [reply src/libraries/restore/runtime/runtime.depproj]: I'd prefer doing that in another PR.

File: src/installer/pkg/projects/Microsoft.NETCore.DotNetHostPolicy/Microsoft.NETCore.DotNetHostPolicy.pkgproj [hunk 21]
```diff
@@ -8,6 +8,12 @@
     <Dependency Include="Microsoft.NETCore.DotNetHostResolver" VersionProp="HostResolverVersion" />
   </ItemGroup>
 
-  <Import Project="$(MSBuildProjectName).props" />
+  <Import Project="$(MSBuildProjectName).props" Condition="'$(TargetOS)' != 'iOS'" />
```
[the remark this answers is not in the document]
Comment (steveisok) [reply src/installer/pkg/projects/Microsoft.NETCore.DotNetHostPolicy/Microsoft.NETCore.DotNetHostPolicy.pkgproj:12]: From what I can tell, `$(PackageTargetRuntime)` has a value, so `$(TargetOS)` or some kind of other check seems to be needed.

File: src/installer/pkg/projects/Directory.Build.targets [hunk 22]
```diff
@@ -14,6 +14,17 @@
     <StableVersion Condition="'$(IncludePreReleaseLabelInPackageVersion)' != 'true'">$(Version)</StableVersion>
   </PropertyGroup>
 
+  <!--
+    For mobile targets, we do not need the host, so allow depproj's to opt 
+    in in order to skip building. 
+  -->
+  <Target Name="MobileSkipBuildProps" BeforeTargets="GetSkipBuildProps" 
+    Condition="'$(SkipBuildOnRuntimePackOnlyOS)' and '$(TargetOS)' == 'iOS'">
```
Comment (dagood) [inline src/installer/pkg/projects/Directory.Build.targets:22]: (Nit) I find this style of wrapping significantly harder to read quickly vs. chopping or alignment. `BeforeTargets` is hard to see, and `Condition` being left of `Name` is unusual. This is a thing that maybe you can get used to, but thought I'd mention it anyway while I'm here and because it's in src/installer.

File: src/installer/pkg/projects/Microsoft.NETCore.DotNetHostPolicy/Microsoft.NETCore.DotNetHostPolicy.pkgproj (hunk 21 above)
[the remark this answers is not in the document]
Comment (dagood) [reply src/installer/pkg/projects/Microsoft.NETCore.DotNetHostPolicy/Microsoft.NETCore.DotNetHostPolicy.pkgproj:12]: Maybe I'm misinterpreting your comment, but yes, that's why I suggested checking for existence of the file. 😄 That would handle the generic case.

**Review by dagood — APPROVED** [review]

File: src/installer/pkg/projects/Directory.Build.targets (hunk 22 above)
Comment (safern) [inline src/installer/pkg/projects/Directory.Build.targets]: Nit, we usually not use boolean conditions based from properties, we usually do direct string comparison ourselves, because boolean condition, forces you to set it to either false or true in all code paths. If we made this a boolean, the default could be false, so sometime like `'$(SkipBuildOnRuntimePackOnlyOS)' != 'true'` then that way you only set it to true when you want to disable this target but not set it to false anywhere.

File: eng/liveBuilds.targets (hunk 14 above)
Comment (akoeplinger) [reply eng/liveBuilds.targets:85]: It looks like `<Choose>` only works at the project level, not inside a Target: https://github.com/Microsoft/msbuild/issues/2056

File: src/installer/pkg/projects/Directory.Build.targets [hunk 23]
```diff
@@ -14,6 +14,18 @@
     <StableVersion Condition="'$(IncludePreReleaseLabelInPackageVersion)' != 'true'">$(Version)</StableVersion>
   </PropertyGroup>
 
+  <!--
+    For mobile targets, we do not need the host, so allow depproj's to opt 
+    in in order to skip building.
```
Comment (akoeplinger) [inline src/installer/pkg/projects/Directory.Build.targets:19]: opt in -> opt out
